### PR TITLE
Fix NPE if map is configured with undefined WAN replication

### PR DESCRIPTION
**Pull request by blazember**
Fixes hazelcast/hazelcast-enterprise/issues/2306

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java [hunk 1]
```diff
@@ -243,11 +243,13 @@ public void initWanReplication(NodeEngine nodeEngine) {
...
         WanReplicationConfig replicationConfig = config.getWanReplicationConfig(wanReplicationRefName);
-        for (WanPublisherConfig publisherConfig : replicationConfig.getWanPublisherConfigs()) {
-            if (publisherConfig.getWanSyncConfig() != null
-                    && ConsistencyCheckStrategy.MERKLE_TREES.equals(publisherConfig.getWanSyncConfig()
-                                                                                   .getConsistencyCheckStrategy())) {
-                return true;
+        if (replicationConfig != null) {
+            for (WanPublisherConfig publisherConfig : replicationConfig.getWanPublisherConfigs()) {
+                if (publisherConfig.getWanSyncConfig() != null
+                        && ConsistencyCheckStrategy.MERKLE_TREES.equals(publisherConfig.getWanSyncConfig()
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java]: seems `publisherConfig.getWanSyncConfig()` can also be null?

Comment (blazember) [reply hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java]: `publisherConfig.getWanSyncConfig()` may return `null`, but it is null-checked before referenced. 
`replicationConfig.getWanPublisherConfigs()` can't return null based on the code, null-check can be added though.
Maybe I miss your point?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java]: no, you are right, i missed that check.

**Comment by blazember** [comment]
run-lab-rub

**Comment by blazember** [comment]
run-lab-run

**Pull request merged**